### PR TITLE
Feature/checkout promotion

### PR DESCRIPTION
**Pull request by mloeffle**
- add form for entering promotion/discount code
- display promotion/discount in line item list
- display line items & reduced summary in offcanvas cart
- refactor summary & sammary values

### CODE REVIEW

File: pages/search/index.vue [hunk 1]
```diff
@@ -32,7 +32,8 @@ setInitialListing(productSearch.value as Schemas['ProductListingResult']);
     <div class="container">
         <h1 class="mb-6 text-center">
             <span v-if="products?.length">
-                results for "<strong>{{ searchTerm }}</strong>"
+                results for "<strong>{{ searchTerm }}</strong
+                >"
```
Comment (aheartforspinach) [inline pages/search/index.vue:36]: formatting seems a little bit off?

File: components/checkout/CheckoutLineItem.vue [hunk 2]
```diff
@@ -125,29 +134,25 @@ const debounceUpdate = useDebounceFn(updateQuantity, 600);
...
-                    v-if="isRemovable"
-                    type="button"
-                    class="bg-transparent text-brand-primary-dark"
-                    :class="{ 'text-gray-medium': isLoading }"
-                    @click="removeCartItem"
-                >
-                    Remove
-                </button>
-            </div>
+            <button
```
Comment (aheartforspinach) [inline components/checkout/CheckoutLineItem.vue:147]: Didnt we have a button component? If not: Would it make sense? Or do we use a FormKitButton here?

File: components/checkout/CheckoutPromotion.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+type FormkitPromotionFields = {
+    promotionCode: string;
+};
+
+const { addPromotionCode, consumeCartErrors } = useCart();
+const { pushError, pushSuccess } = useNotifications();
+
+const cartErrors = ref<Schemas['Cart']['errors']>([]);
+
+const handleSubmit = async (fields: FormkitPromotionFields) => {
```
Comment (aheartforspinach) [inline components/checkout/CheckoutPromotion.vue:13]:
```suggestion
const addPromotion = async (fields: FormkitPromotionFields) => {
```

File: pages/search/index.vue (hunk 1 above)
Comment (mloeffle) [reply pages/search/index.vue:36]: yeah the linter seem to get confused with the strong tag within the double quotes. I moved the double qoutes into the tag in #32

**Review by aheartforspinach — APPROVED** [review]

**Pull request merged**